onnxruntime/core/providers/webnn/builders/impl/qdq_op_builder.cc

```diff
@@ -134,21 +149,11 @@ bool QDQOpBuilder::IsOpSupportedImpl(const GraphViewer&,
     return false;
   }
 
-  // WebNN requires the scale_shape to be a subsample of the input_shape.
   if (scale_shape.size() > input_shape.size()) {
     LOGS(logger, VERBOSE) << "The rank of scale is larger than the rank of input";
     return false;
   }
 
-  for (size_t i = 0; i < scale_shape.size(); ++i) {
-    auto scale_dim = scale_shape[scale_shape.size() - i - 1];
-    auto input_dim = input_shape[input_shape.size() - i - 1];
-    if (input_dim % scale_dim != 0) {
-      LOGS(logger, VERBOSE) << "The shape of scale is not a subsample of the shape of input";
-      return false;
-    }
-  }
-
```

Comment:
Remove this because the ONNX spec already requires the scale shape to be a subsample of the input shape.